TESTING.md

```diff
@@ -0,0 +1,140 @@
+# Geeqie testing and validation patterns
+
```

Comment:
So far as I understand, these tests cannot be run against static functions.
This document:
http://google.github.io/googletest/advanced.html#testing-private-code
describes things that are beyond my comprehension. A recommendation of which path to follow would be good - unless your preference is to not test static functions.

[Maybe you prefer to test only at the public level. Decades ago I did some testing on a project that was itself years old - it worked but required full path coverage for new aircraft. It was an absolute nightmare trying to force a way through each path - even at unit level. I suspect the same would be true for trying to test only at public level on this project]

Reply:
I can add some detail about this (plus an example) in a follow-on PR.  Also, I agree that aspiring to 100% test coverage is often not a great goal.  The goal here is to be able to test the things that make sense to test (I'm talking to you, `FileData`)

But in short, what `static` means is `only visible in this file`.  It's the same as anonymous namespaces in C++, `namespace /*no name here*/ {...}`

So one option is to add the unit tests to the implementation file (either directly, or via some weird include, like `#include "tests/inline_blahblah_test.cc"`).  Another option is to add a trampoline object to the file in question, that will make those static methods visible for unit tests.  This could be done with a macro that is only active when `ENABLE_UNIT_TESTS` is defined.